### PR TITLE
chore: move theme selection to user settings page

### PR DESCRIPTION
**Pull request by DhariniJeeva**
# Description
this PR removes theme selection from menu bar and moves to settings page

 :tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-14452)  

## Screenshots (if appropriate)
Screenshot 2024-08-01 at 9 38 11 AM
Screenshot 2024-08-01 at 9 38 05 AM

## Checklist

- [x] I have added before and after screenshots for UI changes
- ~[ ] I have added JSON response output for API changes~
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/desktop/app/routes/application.js [hunk 1]
```diff
@@ -34,9 +33,13 @@ export default class ApplicationRoute extends Route {
...
     await this.session.setup();
-    const theme = this.session.get('data.theme');
-    this.toggleTheme(theme);
     await this.clusterUrl.updateClusterUrl();
+    const theme = this.session.get('data.theme');
+    /* eslint-disable-next-line ember/no-controller-access-in-routes */
+    const controller = this.controllerFor(
+      'scopes.scope.projects.settings.index',
+    );
+    controller.toggleTheme(theme);
```
Comment (DhariniJeeva) [inline ui/desktop/app/routes/application.js:42]: I had to do this so the color selection is applied when the app refreshes, on refresh we take the user to the landing page. If I were to set this in `settings` route `beforemodel` hook, then after a refresh, the user would only see their selection when they navigate to the settings route which didn't seem like a good UX.  Let me know your thoughts on this approach

Comment (DhariniJeeva) [reply ui/desktop/app/routes/application.js:42]: on second thought, maybe we shouldn't take them to the landing page(targets) on refresh? 🤔 By refresh I mean, a forced manual refresh (cmd+ r)

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js [hunk 2]
```diff
@@ -3,6 +3,68 @@
...
+    value: 'dark',
+  },
+];
+
+export default class ScopesScopeProjectsSettingsIndexController extends Controller {
+  @controller('scopes/scope/projects/settings/index') settings;
+  // Services
+
+  @service session;
+  @service ipc;
```
Comment (ZedLi) [inline ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:31]: Is this ipc service needed?

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js [hunk 3]
```diff
@@ -3,6 +3,68 @@
...
+    value: 'light',
+  },
+  {
+    label: 'dark',
+    value: 'dark',
+  },
+];
+
+export default class ScopesScopeProjectsSettingsIndexController extends Controller {
+  @controller('scopes/scope/projects/settings/index') settings;
```
Comment (ZedLi) [inline ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:27]: What was this for? Doesn't this refer to the same controller?

File: ui/desktop/app/routes/application.js [hunk 4]
```diff
@@ -34,9 +33,13 @@ export default class ApplicationRoute extends Route {
   async beforeModel() {
     this.intl.setLocale(['en-us']);
     await this.session.setup();
-    const theme = this.session.get('data.theme');
-    this.toggleTheme(theme);
     await this.clusterUrl.updateClusterUrl();
+    const theme = this.session.get('data.theme');
+    /* eslint-disable-next-line ember/no-controller-access-in-routes */
+    const controller = this.controllerFor(
```
Comment (ZedLi) [inline ui/desktop/app/routes/application.js:39]: Hmm I don't think we should reach down into a child controller. I'm thinking the `toggleTheme` action you have in the controller should either be in a service or in this `application` controller and then you call it from the settings controller

Comment (DhariniJeeva) [reply ui/desktop/app/routes/application.js]: thanks for the feedback, I moved it to app controller

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js (hunk 3 above)
Comment (DhariniJeeva) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js]: yup this wasn't needed, I forgot to remove this

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js (hunk 2 above)
Comment (DhariniJeeva) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js]: thanks, not needed, removed!

**Review by ZedLi — APPROVED** [review]
Looks good!

**Pull request merged**